src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -38,6 +39,20 @@ public static IResourceBuilder<PostgresServerResource> AddPostgres(this IDistrib
         var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
 
         var postgresServer = new PostgresServerResource(name, userName?.Resource, passwordParameter);
+
+        builder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>(async (@event, ct) =>
+        {
+            var connectionString = await postgresServer.GetConnectionStringAsync(ct).ConfigureAwait(false);
+            builder.Configuration[$"{postgresServer.Name}_builtin_connectionstring"] = connectionString;
+        });
+
+        builder.Services.AddHealthChecks().AddNpgSql(sp =>
+            {
+                return builder.Configuration[$"{name}_builtin_connectionstring"]!;
+            },
+            name: $"{name}_check"
+        );
```

Comment:
What does using configuration do here over using a variable in the closure?

Reply:
Yeah I cut over to capturing the variable from the enclosing scope.